packages/ra-supabase-ui-materialui/src/ForgotPasswordForm.tsx

```diff
@@ -33,10 +33,14 @@ export const ForgotPasswordForm = () => {
         },
     });
 
-    const submit = (values: FormData) => {
-        return resetPassword({
-            email: values.email,
-        });
+    const submit = async (values: FormData) => {
+        try {
+            await resetPassword({
+                email: values.email,
+            });
+        } catch (error) {
+            notify(error?.message ?? 'ra.notification.http_error');
+        }
```

Comment:
Not 100% sure but since you have to handle exceptions here, do you still need the `onError` side effect?

Also, the `onError` implementation seems much more complete than this one. Which one is right?

Reply:
You're right, especially as they won't be called at all when using `mutateAsync`